udata/harvest/backends/dcat.py

```diff
@@ -59,13 +61,18 @@ def extract_graph(source, target, node, specs):
 class DcatBackend(BaseBackend):
     display_name = 'DCAT'
 
-    def initialize(self):
-        '''List all datasets for a given ...'''
+    def inner_harvest(self):
         fmt = self.get_format()
-        graphs = self.parse_graph(self.source.url, fmt)
-
         self.job.data = { 'format': fmt }
 
+        graphs = self.walk_graph(
```

Comment:
I think this logic/lahout hides a bit the fact that we're processing the graph, not just creating it (process_datasets being hidden).

Reply:
I may be my functional programmer mind :-D Walking a tree is a known operation for me :-)

To fix this, what do you prefer:
1. Add a comment above explaining what we are doing
2. Rename the method, maybe something like `walk_graph_and_process`?
3. Having new functions without the callback (but I need it for dataservices so I don't have right now a solution to allow to switch between the two modes…)

Reply:
Used `yield` inside 064f8eb3c. Much better! Thanks!